packages/plugin-vue-jsx/src/index.ts

```diff
@@ -53,55 +53,50 @@ function vueJsxPlugin(options = {}) {
           sourceFileName: id
         })
 
+        if (!result) {
+          return {
+            code: '',
+            map: ''
+          }
+        }
+
         if (!needHmr) {
           return {
-            code: result.code,
-            map: result.map
+            code: result.code!,
+            map: result.map!
           }
         }
 
         // check for hmr injection
-        /**
-         * @type {{ name: string }[]}
-         */
-        const declaredComponents = []
-        /**
-         * @type {{
-         *  local: string,
-         *  exported: string,
-         *  id: string,
-         * }[]}
-         */
-        const hotComponents = []
+        const declaredComponents: { name: string }[] = []
+        const hotComponents: {
+          local: string
+          exported: string
+          id: string
+        }[] = []
         let hasDefault = false
 
-        for (const node of result.ast.program.body) {
-          if (node.type === 'VariableDeclaration') {
-            const names = parseComponentDecls(node, code)
+        traverse(result.ast, {
```

Comment:
I intentionally didn't use `traverse` here because export declarations can only appear at root level, so a direct iteration is faster.